### DISCUSSION TITLE
开发环境搭建的踩坑记录: npm install 失败 + docker-compose up 之后连不上MongoDB等

### DISCUSSION THREAD

**Discussion by cccy0** [labring/laf#195] [Feedbacks]
### npm install 失败
1. 必须使用npm官方源, 不建议使用taobao源
2. 若在linux环境下用root用户操作, 则必须使用 `npm install --unsafe-perm`
3. 若出现Cannot read properties of null (reading 'pickAlgorithm'), 则需要定位到是哪一个package出现问题, 然后在package所在package.json目录下执行`npm cache clear --force`
### docker-compose up 之后连接MongoDB超时
1. 在开发环境的docker-compose.yml文件中, system-server服务下, SYS_DB_URI和APP_DB_URI两个环境变量后面都加上`&directConnection=true`
   
如果有别的坑会在此下面继续补充

**Comment by maslow** [comment]
按我理解，npm cache clear --force 应该在任意目录下运行都可以，这个失败应该是因为之前有不正确的包缓存到本地了，清除从官方源重新 install 即可。

**Comment by maslow** [comment]
`directConnection=true`
这个属于 mongodb 的范畴，不适合 使用这个参数，特别是生产环境。

连接不上，可能是 你修改了 mongodb uri 中的 host，与 mongodb server host 不一致，或其它原因， 这个是问题， 非开发环境 不应该使用 `directConnection=true` 这个参数。

**Reply by maslow** [reply]
补充说明： @cccy0 后续反馈，这个问题只出现在 wsl2 环境中
